tests/suites/test_suite_debug.function

```diff
@@ -89,10 +95,15 @@ void mbedtls_debug_print_ret( char * file, int line, char * text, int value,
     memset( buffer.buf, 0, 2000 );
     buffer.ptr = buffer.buf;
 
-    TEST_ASSERT( mbedtls_ssl_setup( &ssl, &conf ) == 0 );
+    mbedtls_ssl_config_defaults( &conf,
+                                 MBEDTLS_SSL_IS_CLIENT,
+                                 MBEDTLS_SSL_TRANSPORT_STREAM,
+                                 MBEDTLS_SSL_PRESET_DEFAULT );
```

Comment:
@mpg FYI this seems to be a bug in `test_suite_debug`, in two ways:
* It modifies the config after `mbedtls_ssl_setup()`
* It doesn't setup the config properly before calling `mbedtls_ssl_setup()`, e.g. via `mbedtls_ssl_config_defaults()`.
We're running into this only now that we begin to perform sanity checks on the config in `mbedtls_ssl_setup()`.